backend/src/features/recipes/recipes.controller.ts

```diff
@@ -5,17 +5,61 @@ import { PageList } from "../../models/PageData";
 import { AddRecipesDto } from "./dtos/AddIngredients.dto";
 import { IngredientDetailsDto } from "../ingredients/dtos/IngredientDetailsDto";
 import { TagDetailsDto } from "../tags/dtos/TagDetailsDto";
+import { IOrderBy } from "../../types/IOrderBy";
 
 interface RequestQuery {
     page: number,
     pageSize: number,
-    search?: string
+    search?: string,
+    orderBy?: string
 }
 
-
 export const getRecipes = async (req: Request<{}, {}, {}, RequestQuery>, res: Response) => {
-    const {page = 1, pageSize = 10, search} = req.query
-    const [results, totalItemsCount] = await recipesRepository.getRecipes(page, pageSize, search);
+    const {page = 1, pageSize = 10, search, orderBy} = req.query
+
+    let orderByObj: IOrderBy[] = [];
+    if (orderBy != undefined) {
+        switch (orderBy) { 
+            case "titleAsc":
+                orderByObj.push({
+                    title: "asc"
+                });
+                break;
+
+            case "titleDesc":
+                orderByObj.push({
+                    title: "desc"
+                });
+                console.log("ok");
+                break;
+
+            case "createdAsc":
+                orderByObj.push({
+                    created_at: "asc"
+                });
+                break;
+
+            case "createdDesc":
+                orderByObj.push({
+                    created_at: "desc"
+                });
+                break;
+
+            case "updatedAsc":
+                orderByObj.push({
+                    updated_at: "asc"
+                });
+                break;
+
+            case "updatedDesc":
+                orderByObj.push({
+                    updated_at: "desc"
+                });
+                break;
+        }
+    }
+    
+    const [results, totalItemsCount] = await recipesRepository.getRecipes(page, pageSize, orderByObj, search);
```

Comment:
1. why would we have nullable `orderBy` prop int the object? and check if it's undefined? Example here `if (orderBy != undefined)`'
2. We're not entirely sure if the frontend supports multiple sorting, but looking at this code, it looks like it only support single filtering which is great. 
3. `switch`  statement can be separated into a  method. since the responsibility of this will only return columns on it's own context (which is Recipe for example). with this code,  we can default it to `id`, if the entered order does not exist on the switch case.
```typescript
const getOrder = (order?: string) => {
    switch (order) {
        case "title":
            return "title"
        case "created_at":
            return "created_at"
        default:
            return "id"
    }
}
```
4. We can make an `if` statement if the "Ordering" is `desc` or `asc`, and instead of Array, we can just use basic object, since prisma `orderBy` prop is more object like
```typescript
let orderBy : Record<string, any> = {}
if(sortOrder === "desc")
	// we invoke the `getOrder` func here to get the columns
	orderBy[getOrder(sortColumn)] = "desc"
else
	// we pass `asc` here since if the user don't pass in the sortColumn and sortOrder prop in the req.query, this will be the default, which is id = `asc`. (this could change in the future.)
	orderBy[getOrder(sortColumn)] = "asc"



// and we can pass it the prisma like this

await client.recipe.findMany(
{
	where: {
		id
	},
	orderBy // like this as an object
})

```
5. By doing the 2 scenario above, our RequestQuery type will be something like this
```typescript
interface RequestQuery {
    page: number,
    pageSize: number,
    search?: string,
    sortColumn?: string, // column that user wants to sort
    sortOrder: string  // ordering method which will take "desc" or "asc"
}

const getRecipes = (req: Requst<{}, {}, {}, RequestQuery>) => {
	const {sortColumn, sortOrder} = req.query;

	//... do the above code below.
}

```